backend/src/main/java/gov/cdc/usds/simplereport/api/patientExperience/PatientExperienceController.java

```diff
@@ -0,0 +1,123 @@
+package gov.cdc.usds.simplereport.api.patientExperience;
+
+import java.util.Map;
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import static gov.cdc.usds.simplereport.api.Translators.parseSymptoms;
+import static gov.cdc.usds.simplereport.api.Translators.parseEmail;
+import static gov.cdc.usds.simplereport.api.Translators.parseEthnicity;
+import static gov.cdc.usds.simplereport.api.Translators.parseGender;
+import static gov.cdc.usds.simplereport.api.Translators.parsePhoneNumber;
+import static gov.cdc.usds.simplereport.api.Translators.parseRace;
+import static gov.cdc.usds.simplereport.api.Translators.parseState;
+import static gov.cdc.usds.simplereport.api.Translators.parseString;
+
+import gov.cdc.usds.simplereport.api.exceptions.FeatureFlagDisabledException;
+import gov.cdc.usds.simplereport.api.model.AoEQuestions;
+import gov.cdc.usds.simplereport.api.model.pxp.PxpApiWrapper;
+import gov.cdc.usds.simplereport.api.model.pxp.PxpPersonWrapper;
+import gov.cdc.usds.simplereport.db.model.Organization;
+import gov.cdc.usds.simplereport.db.model.PatientLink;
+import gov.cdc.usds.simplereport.db.model.Person;
+import gov.cdc.usds.simplereport.db.model.TestEvent;
+import gov.cdc.usds.simplereport.db.model.auxiliary.TestResult;
+import gov.cdc.usds.simplereport.service.PatientLinkService;
+import gov.cdc.usds.simplereport.service.PersonService;
+import gov.cdc.usds.simplereport.service.TestEventService;
+import gov.cdc.usds.simplereport.service.TestOrderService;
+
+@RestController
+@RequestMapping(value = "/pxp")
+@Validated
+public class PatientExperienceController {
+  @Autowired
+  private PersonService ps;
+
+  @Autowired
+  private PatientLinkService pls;
+
+  @Autowired
+  private TestOrderService tos;
+
+  @Autowired
+  private TestEventService tes;
+
+  @Value("${feature-flags.patient-links:false}")
+  private boolean patientLinksEnabled;
+
+  /**
+   * This endpoint is used to determine whether or not the patient link is still
+   * valid and has not yet expired. It does not require birthdate verification
+   */
+  @GetMapping("/link/{plid}")
+  public Organization getPatientLinkCurrent(@PathVariable("plid") String internalId) throws Exception {
+    if (!patientLinksEnabled) {
+      throw new FeatureFlagDisabledException("Patient links not enabled");
+    }
+    return pls.getPatientLinkCurrent(internalId);
+  }
```

Comment:
I'd prefer to get rid of this operation entirely and only send in the patient link ID when it and the birthdate are being validated, but I don't know how design would feel about removing the organization name from the form.

Reply:
It also gives us the opportunity to tell the user that their link has expired before the enter their birthdate, but I don't think that buys us anything necessarily! 

This current implementation also requires that patient link id & dob are sent _along with every subsequent request_, but this is a naive first-pass approach.

My intent is to instead respond to a successful plid+dob validation with a new guid, stored in a new column on the PatientLink entity, which the client would then provide as a header with each request. Ideally I'll add a method (or class-level) annotation that can enforce the presence of this header. In a perfect world I'd do it with a Spring security provider, but that looks like a consider lift to implement

Reply:
@jeskew-gov I'll follow up w/ design tomorrow WRT removing the organization name from the DOB verification form

Reply:
@jeskew-gov This endpoint has been removed, and I'm now exposing Organization Name via `PxpPersonWrapper`, dispatching an `updateOrganization` upon successful DOB validation.

The landing screen does not show anything for the org name, as discussed w/ the designers